addons/stock/models/stock_move_line.py

```diff
@@ -92,6 +92,14 @@ def check_reserved_done_quantity(self):
     @api.onchange('product_id', 'product_uom_id')
     def onchange_product_id(self):
         if self.product_id:
+            if self.user_has_groups('stock.group_stock_multi_locations') and not self.id:
```

Comment:
switch parts of the if
if not self.id and self.user_has_groups('stock.group_stock_multi_locations')
shoudlnt you just call _get_putaway_strategy?